practice/react-practice-one/src/components/Pagination/Pagination.jsx

```diff
@@ -0,0 +1,55 @@
+/* Import dependencies */
+import PropTypes from 'prop-types';
+
+/* Import components */
+import Button from '../Buttons/Button';
+
+/* Import CSS */
+import './pagination.css';
+
+const Pagination = ({ currentPage, totalRecords, pageLimit, onPageChange }) => {
+  const totalPages = Math.ceil(totalRecords / pageLimit);
+  const pageRange = 5;
+
+  const handleClick = (page) => {
+    if (page !== currentPage) {
+      onPageChange(page);
+    }
+  };
+
+  const getPageNumbers = () => {
+    const startPage = Math.max(1, currentPage - Math.floor(pageRange / 2));
+    const endPage = Math.min(totalPages, startPage + pageRange - 1);
+
+    const pages = [];
+    for (let page = startPage; page <= endPage; page++) {
+      pages.push(page);
+    }
+
+    return pages;
+  };
+
+  const renderButtons = () => {
+    const pages = getPageNumbers();
+
+    return pages.map((page) => (
+      <Button
+        key={page}
```

Comment:
It is the same as the previous code, you should read the article again and ensure you understand why we should avoid using key as index(1, 2, 3...)

Reply:
I done